api/course.js

```diff
@@ -4,19 +4,63 @@
 require("path");
 const express = require("express");
 const app = express();
-const {getCourse} = require("../models/course");
+const {getCourse, getLiveCourses} = require("../models/course");
+const {getUserById} = require("../models/user");
+const {Role} = require("../entities/role");
 const {requireAuth} = require("../services/auth/auth");
 
+// Get all courses from the Course API server,
+// then process the collected courses so that they conform with our database.
+// Then add each course to the database that doesn't already exist.
+// If the course already exists we update it instead.
+app.get("/updateDatabase", requireAuth, async (req, res) => {
+
+  try {
+
+    console.log("Get all courses from the OSU course API");
+
+    // get user data
+    const authenticatedUser = await getUserById(req.auth.userId);
+
+    // due to the computational cost of this action,
+    // only the head advisor can perform this action
+    if (authenticatedUser.role === Role.advisor ||
+        authenticatedUser.role === Role.headAdvisor) {
+
+      // start course update
+      getLiveCourses(req.auth.userId);
```

Comment:
`await`?

Reply:
So the reason for no await is this: 

The operation is really slow, so slow (possibly hours) that it doesn't make sense to wait for the operation to finish to send a response (most requests will time out before we can get a response back). Instead the expected behavior is that we send a response to the user letting them know that we got their request and we are starting an update.

To let the user know the status of update we instead use notifications to signal when the update is done, this circumvents the issue of a request timing out after waiting for hour(s).